vimeo-networking/src/main/java/com/vimeo/networking/model/VideoFile.java

```diff
@@ -24,21 +24,22 @@
 
 import com.google.gson.annotations.SerializedName;
 import com.vimeo.stag.UseStag;
+import com.vimeo.stag.UseStag.FieldOption;
 
+import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 
 import java.io.Serializable;
 import java.util.Date;
 
 /**
- * Representation of a Video stream/playback file
+ * Base representation of a Video stream/playback file
  * <p>
  * Created by alfredhanssen on 4/25/15.
  */
 @SuppressWarnings("unused")
-// TODO: Remove the VideoFileDeserializer and use Stag instead, once API corrects log issue behind version 2/16/17 [KZ]
-// @UseStag(FieldOption.SERIALIZED_NAME)
-public class VideoFile implements Serializable {
+@UseStag(FieldOption.SERIALIZED_NAME)
+public abstract class VideoFile implements Serializable {
```

Comment:
Keeping `VideoFile` not abstract and as a single class (rather than subclassing it for dash/hls/progressive) definitely worked for us in the past, especially when we received all of the items in a single array. I think specialized VideoFile classes add benefit, now that we receive specialized representations within the Play response. Dash and HLS files do not receive the information contained within the progressive response, like size, width, height, md5, mimetype, etc. Likewise, Dash contains extra stuff for DRM that none of the other receive. And whenever we access a `VideoFile` we would have to compare it to a location on the parent Play object to determine which file type it was. This was my rationale for this change.

We can easily do a `instanceof` check or use `getQuality()` to determine the file type now. Since these DTOs are created through deserialization using Stag/Gson, in order to properly have them know their quality without receiving it in JSON, I created concrete methods that are on the subclasses. 

Do you have a suggestion on how to accomplish this another way?